plugins/scaffolder-react/src/next/components/ScaffolderPageContextMenu/ScaffolderPageContextMenu.tsx

```diff
@@ -100,7 +106,7 @@ export function ScaffolderPageContextMenu(
               <ListItemText primary="Create" />
             </MenuItem>
           )}
-          {onEditorClicked && (
+          {onEditorClicked && canManageTemplates && (
```

Comment:
mmm this won't stop people from manually invoked the endpoints, right?
Is this intended?

Reply:
The use of `<RequirePermission>` added in `Router.tsx` should prevent the UI routes from being accessed. And from what I can see none of the template management features make any calls to backend scaffolder endpoints that would need to be protected by a permission check. The exception with this is the dry-runner which does make a call to the `/v2/dry-run` endpoint that is protected with the `taskCreatePermission`:

https://github.com/backstage/backstage/blob/23ec9a8d11e32163f9ab1e6073fcd56512f01d01/plugins/scaffolder-backend/src/service/router.ts#L760-L766

Does it make sense to add the new `templateManagementPermission` to the list of required permissions for the dry run endpoint?

Reply:
ah I see!
What about using the `taskCreatePermission` here instead of introducing a new `templateManagementPermission`? Could make sense?

Reply:
I don't know if that makes sense because `taskCreatePermission` is associated with executing tasks/templates. In my case, I want to forbid users from using the task management features yet still allow them to execute templates.

Reply:
I agree with @stephenglass in this case. These are two different scenarios which can't be solved with the same permission.

Reply:
totally agree!